### PR TITLE
t/153: List items should handle Enter and Space key press when focused

### LINKED ISSUE #153: List component doesn't handle the Enter and Space keys
feb-02-2017 10-51-47

Enter and space keys on buttons are handled, so should be in this case too.

BTW, there's also that issue that the UI events' callbacks don't focus the editor before executing commands. Perhaps, we should work on all that together.

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: List items should handle Enter and Space key press when focused. Closes #153.

Requires: ckeditor/ckeditor5-utils#169.

### CODE REVIEW

**Comment by oleq** [comment]
https://github.com/ckeditor/ckeditor5-ui/pull/265/commits/98629db52f5b3fd7fedba01bd5474a6b3ec8f5d5 implements the whole thing in the `Template`. https://github.com/ckeditor/ckeditor5-ui/pull/265/commits/7deb13b5851499ead7ea91d809cad8f3a7dfb114 uses `KeystrokeHandler`, which is IMO a better choice because it separates us from native DOM API (like `preventDefault`) and will be safer in the future as KH gets improved and secured.

File: src/list/listitemview.js [hunk 1]
```diff
@@ -93,9 +102,34 @@ export default class ListItemView extends View {
...
 		this.element.focus();
 	}
 }
+
+// Returns the Enter and Space keystroke handler for given ListItemView.
+//
+// @private
+// @param {module:ui/list/listitemview~ListItemView} view
+// @returns {Function} A listener for {@link module:utils/keystrokehandler~KeystrokeHandler}.
+function getKeystrokePressCallback( view ) {
```
Comment (oskarwrobel) [inline src/list/listitemview.js]: Why not a simple local function inside init() method?
```javascript
const keystrokePressCallback = ( data, cancel ) => {
	this.fire( 'execute' );
	cancel();
};
```

File: src/list/listitemview.js [hunk 2]
```diff
@@ -93,9 +102,34 @@ export default class ListItemView extends View {
...
+	 * @inheritDoc
+	 */
+	init() {
+		const onKeystrokePress = getKeystrokePressCallback( this );
+
+		this.keystrokes.listenTo( this.element );
+
+		// Execute on Enter and Space key press.
+		this.keystrokes.set( 'Enter', onKeystrokePress );
+		this.keystrokes.set( [ 32 ], onKeystrokePress );
```
Comment (oskarwrobel) [inline src/list/listitemview.js]: I'm wondering why <kbd>Space</kbd> is not defined in our `keyCodes`.

Comment (oleq) [inline src/list/listitemview.js]: Me too TBH. I'll create a PR for this.

**Pull request merged**